test/__snapshots__/test/test.ts.md

```diff
@@ -1861,6 +1861,92 @@ Generated by [AVA](https://avajs.dev).
     }␊
     `
 
+## intersection.4.js
+
+> Expected output to match snapshot for e2e test: intersection.4.js
+
+    `/* eslint-disable */␊
+    /**␊
+     * This file was automatically generated by json-schema-to-typescript.␊
+     * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,␊
+     * and run json-schema-to-typescript to regenerate this file.␊
+     */␊
+    ␊
+    export type Intersection = A | B;␊
+    export type A = A1 & {␊
```

Comment:
I think I missed this earlier -- this is a case where the output in #605 is better than #603. I wonder if there's a way to improve this output, so we don't generate `A1` and `B1`, and instead just use `Base` directly.

Like I did here: https://github.com/bcherny/json-schema-to-typescript/pull/605/files#diff-34615f003eb49f90c3044b8fafebe3fe6b162e290b9b724082a6d72f58cfa4afR1876

Reply:
Thanks! I took a try at that in 7d3f495 by pulling schema's pre-existing `allOf` rule into the intersection. Would you mind taking a third pass and let me know if I missed anything?